### PR TITLE
Scrum 72 post > email > draft: Refactor email draft POST tests and enhance teardown functionality

### PR DESCRIPTION
**Pull request by Diego-Claros17**
This pull request focuses on refactoring the email draft POST tests and ensuring the teardown functionality operates correctly. The changes made enhance the code structure and ensure reliable cleanup after tests.

Changes Made:

1. Refactored POST Email Draft Tests:
  
   - Included comprehensive payloads (`payload_success` and `payload_invalid_email_format`) for various test scenarios to ensure consistency and clarity in test cases.

2.  Enhanced Teardown Functionality:
3. 
   - Ensured that the teardown process correctly cleans up email drafts created during tests, preventing test data from persisting between test runs.
   - Validated that the teardown successfully deletes all drafts, confirming its reliability and effectiveness.

Detailed Changes

- test_create_email_draft_success: Verifies that a draft email can be successfully created and receives a status code of 201.
- test_create_email_draft_missing_fields: Ensures that creating a draft with missing fields returns a status code of 400.
- test_create_email_draft_invalid_email_format: Tests that using an invalid email format results in a status code of 400.
- Teardown Process: Implemented checks to verify the successful deletion of drafts after each test, with detailed logging for better traceability.

Testing

- All refactored tests have been executed successfully.
- The teardown process has been thoroughly tested to confirm it effectively cleans up after each test.

### CODE REVIEW

File: api_endpoints/mail_draft_endpoints.py
Comment (AlisonGuzman2006) [inline api_endpoints/mail_draft_endpoints.py:1]: The code additions have been accepted

File: config/config.py
Comment (AlisonGuzman2006) [inline config/config.py:1]: change approved

File: payloads/mail_draft/payload_email_draft_invalid_email_format.json
Comment (AlisonGuzman2006) [inline payloads/mail_draft/payload_email_draft_invalid_email_format.json:1]: code addition approved

File: payloads/mail_draft/payload_email_draft_success.json
Comment (AlisonGuzman2006) [inline payloads/mail_draft/payload_email_draft_success.json:1]: approved the code additions

**Review by Jasonn5 — APPROVED** [review]
Reviewed, excecuted, good job

File: tests/mail_draft/conftest.py
Comment (AlisonGuzman2006) [inline tests/mail_draft/conftest.py:1]: approved

File: tests/mail_draft/test_get_email_draft.py
Comment (AlisonGuzman2006) [inline tests/mail_draft/test_get_email_draft.py:1]: approved

File: tests/mail_draft/test_post_email_draft.py
Comment (AlisonGuzman2006) [inline tests/mail_draft/test_post_email_draft.py:1]: the addition of the code has been approved

File: utils/load_resources.py
Comment (AlisonGuzman2006) [inline utils/load_resources.py:1]: the code addition has been approved

**Review by AlisonGuzman2006 — APPROVED** [review]
The changes and code additions have been reviewed and approved

**Pull request merged**